src/pineko/cli/fonll.py

```diff
@@ -0,0 +1,163 @@
+    ):
+        raise InconsistentInputsError(
+            "One between ffn04 and ffns5til has been provided without the other. Since they are both needed to construct FONLL, this does not make sense."
+        )
+    if (ffn03 is None and (ffns4til is not None or ffns4 is not None)) or (
+        ffn03 is not None and (ffns4til is None and ffns4 is None)
+    ):
+        raise InconsistentInputsError(
+            "One between ffn03 and ffns4til has been provided without the other. Since they are both needed to construct FONLL, this does not make sense."
+        )
+
+    # check that if we have ffns we dont have any bar or til
+    if ffns4 is not None or ffns5 is not None:
+        if not all([fk is None for fk in [ffns4til, ffns4bar, ffns5til, ffns5bar]]):
+            raise InconsistentInputsError(
+                "If ffns4 and ffns5 are provided, no ffnstil or ffnsbar should be provided."
+            )
+
+    # Get theory info
+    tcard = theory_card.load(theoryid)
+    if not "DAMPPOWER" in tcard:
+        if tcard["DAMP"] != 0:
+            raise InconsistentInputsError
+        tcard["DAMPPOWER"] = None
+    # Getting the paths to the grids
+    grids_name = get_list_grids_name(
+        configs.configs["paths"]["ymldb"] / f"{dataset}.yaml"
+    )
+    for grid in grids_name:
+        # Checking if it already exists
+        new_fk_path = configs.configs["paths"]["fktables"] / str(theoryid) / grid
+        if new_fk_path.exists():
+            if not overwrite:
+                rich.print(
+                    f"[green]Success:[/] skipping existing FK Table {new_fk_path}"
+                )
+                return
+        fonll.produce_combined_fk(
+            *(
+                load_cfg(str(name), grid)
+                for name in (
+                    ffns3,
+                    ffn03,
+                    ffns4,
+                    ffns4til,
+                    ffns4bar,
+                    ffn04,
+                    ffns5,
+                    ffns5til,
+                    ffns5bar,
+                )
+            ),
+            theoryid,
+            damp=(tcard["DAMP"], tcard["DAMPPOWER"]),
+        )
+        if new_fk_path.exists():
+            rich.print(f"[green]Success:[/] Wrote FK table to {new_fk_path}")
+        else:
+            rich.print(f"[red]Failure:[/]")
+
+
+@command.command("fonll_tcards")
+@click.argument("theoryID", type=int)
+@config_setting
+def fonll_tcards(theoryid, cfg):
+    """Produce the FONLL tcards starting from the original tcard given by the theoryID."""
+    path = configs.detect(cfg)
+    base_configs = configs.load(path)
+    configs.configs = configs.defaults(base_configs)
+    if cfg is not None:
+        print(f"Configurations loaded from '{path}'")
+    tcard = theory_card.load(theoryid)
+    tcard_parent_path = theory_card.path(theoryid).parent
+    if "FONLL" not in tcard["FNS"]:
+        raise TheoryCardError
```

Comment:
Do you really need this check here?

To avoid proliferating code it is useful to check as late as possible (or even better: never - just pretend it's working, until it will raise an error on its own)

Reply:
Yes but, according to what is `tcard["FNS"]`, the error raised could be rather difficult to understand. For this reason I believe it is better to have an explicit error at least for this fundamental configuration, i.e. that the `FNS` is of `FONLL` kind.